tokio/src/task/join_set.rs

```diff
@@ -374,6 +374,26 @@ impl<T: 'static> JoinSet<T> {
         while self.join_next().await.is_some() {}
     }
 
+    /// Awaits the completion of all tasks on this `JoinSet`, returning a Vector of their results.
+    ///
+    /// If any task on this `JoinSet` errors or panics, this will panic and cancel all
+    /// tasks on this `JoinSet`.
+    ///
+    /// For specific error handling of individual tasks, you should call `join_next`
+    /// in a loop with your desired handling.
```

Comment:
I think the wording could be improved a bit. For example:
```suggestion
    /// This a convenience method that is equivalent to calling [`join_next`] in
    /// a loop. If any tasks on the `JoinSet` fail with an error, then this call
    /// to `join_all` will panic. To handle panics in any other way, manually
    /// call `join_next` in a loop.
    ///
    /// [`join_next`]: Self::join_next
```
It would also be nice with an example. For example, you could adjust one of the examples from the top of this file to use `join_all`. I also think we should have an example that shows a manual loop using `join_next`, explaining that `join_all` is equivalent to that loop.

Reply:
Oh, also, the documentation should explain that the elements will be stored in the order in which they finish. And *not* the order in which they were spawned.